samples/saga/batching/sample.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Batch messages using Saga
+summary: Use batching when generating tasks using Saga
+reviewed: 2022-04-09
+component: Core
+related:
+- nservicebus/sagas
+---
+
+
+## Introduction
+
+This sample shows how to control the task (work) generation via batching of the work items through Sagas.
```

Comment:
1. Its missing why someone would want to do this.
2. Its missing alternatives. Like, do processing on an isolated endpoint or even a [sending work items to a specific endpoint instance](https://docs.particular.net/nservicebus/messaging/routing#make-instance-uniquely-addressable) where this endpoint could have a specific lower concurrency or even [processing rate limit](https://github.com/ramonsmits/NServiceBus.RateLimiter).
3. The example isn't stating how to do the paging.
4. A drawback of this solution is that not all work items are processed as fast as possible but I guess that is the whole reason for this overhead to throttle?
5. As critical time can no longer be used to validate to "peek" workload it might be worth stating that it likely is very good idea to capture the time delta between saga start and completion to track how long this "batch operation" takes in total and emit this via a metric framework.
6. An orchestration like alternative could also be based on a routing slip to achieve sequential atomic work item processing without the saga overhead IO to significantly reduced wasted IO and latency.

Reply:
I have updated, and hopefully addressed points 1, 2, and 5. Regarding the other points:

3. The sample is abstract to show the concept. The paging depends on the type of work that needs to be generated.
4. All generated work items are processed as fast as possible. If multiple work items are causing messages to generate, that'd cause massive queue backup which wouldn't happen with this pattern (since we only generate them gradually). E.g. Customers A and B and C generate 3 work tasks. If we generate 10,000 messages per customer, the first customer coming in would cause massive delays for customers B and C.
5. Correct. A feature such as a sequence diagram (if it could show the whole picture) or an activity view via open telemetry is a better fit for distributed systems to show / measure end-to-end time. 
6. Do we have a sample to point to as another alternative?

Reply:
> 6\. Do we have a sample to point to as another alternative?

We have a routing slip sample: https://docs.particular.net/samples/routing-slips/

Reply:
Thanks, added.